DS13/code/modules/EMH/EMH.dm

```diff
@@ -0,0 +1,291 @@
+	if(istype(user, /mob/living/carbon/human/species/holographic)) //Allow them to jump to different locs.
+		var/list/jumplocs = list()
+		for(var/obj/machinery/emh_emitter/EM in GLOB.machines)
+			if(istype(EM))
+				jumplocs += EM
+		var/A
+		A = input("Transfer program to where?", "EMH subsystem", A) as null|anything in jumplocs
+		if(!A)
+			return
+		user.say("Computer, transfer emergency medical holographic program to [get_area(A)]")
+		new /obj/effect/temp_visual/dir_setting/ninja/cloak(get_turf(user), user.dir)
+		user.forceMove(get_turf(A))
+		playsound(user.loc, 'DS13/sound/effects/holofizzle.ogg', 100, 0)
+		var/obj/machinery/emh_emitter/S = A
+		emh = null
+		S.emh = user //transfer silently.
+		return
+	if(is_occupied()) //If we have an EMH, or there is an EMH in the world
+		to_chat(user, "There is already an EMH active, summon it?")
+		var/question = alert("Summon an EMH?",name,"yes","no")
+		if(question == "yes")
+			var/mob/living/carbon/human/species/holographic/S = locate(/mob/living/carbon/human/species/holographic) in GLOB.alive_mob_list
+			if(S)
+				var/emhconsent = alert("You have been summoned by [user] in [get_area(src)]. Transfer to their location?",name,"accept","reject")
+				if(emhconsent == "accept")
+					activate(S)
+					user.say("Computer, transfer emergency medical holographic program to [get_area(src)]")
+				else
+					to_chat(user, "The EMH has declined your summons.")
+					return ..()
+	else
+		if(emh)
+			emh.death()
+		if(user)
+			user.say("Computer, activate emergency medical hologram")
+			to_chat(user, "Attempting to activate EMH ((This requires a ghost willing to control it...))")
+		activate(null) //No EMH present, try to make one!
+
+/obj/machinery/emh_emitter/process()
+	var/area/A = get_area(src) //If someone moves us, ensure we update name.
+	name = "EMH emitter ([A])"
+	if(QDELETED(emh))
+		emh = null
+		icon_state = "emh-off"
+		return
+	if(!is_occupied()) //Do we HAVE an EMH? If not, stop processing and turn off.
+		return
+	if(emh in orange(src, range)) //Orange ignores walls, so our EMH has a tiny bit of movement freedom
+		icon_state = "emh-on"
+		return
+	else
+		var/obj/machinery/emh_emitter/S = locate(/obj/machinery/emh_emitter) in get_area(emh)
+		if(S)
+			if(emh in orange(S, S.range))
+				S.emh = emh //Transfer his program seamlessly
+				emh = null
+				icon_state = "emh-off"
+		else
+			to_chat(emh, "You cannot move beyond the range of your holo-emitters!")
+			emh.forceMove(get_turf(src))
+
+/obj/machinery/emh_emitter/proc/activate(var/mob/living/carbon/human/species/holographic/transferred = null) //Activate, if we're not transferring an existing EMH over, then make a new one!
+	if(transferred)
+		emh = transferred
+		new /obj/effect/temp_visual/dir_setting/ninja/cloak(get_turf(emh), emh.dir)
+		transferred.forceMove(get_turf(src))
+		to_chat(transferred, "You have been transferred to [get_area(src)]")
+		START_PROCESSING(SSmachines,src)
```

Comment:
You have two of these in this proc. Which one is redundant? And which one is incorrectly placed here?